### ISSUE TITLE
Duration apparently wrong

### ISSUE THREAD

**Issue by mikemajara** [date-fns/date-fns#2543]
```
formatDuration(intervalToDuration({
  end: new Date(2021, 8, 1, 12, 0, 0), 
  start: new Date(),
  
}), {delimiter: ", "})
// output: 2 months, 2 hours, 56 minutes, 56 seconds
```

Today is 1st of July. In a month time, it will be 1st of August. So I assume that `intervalToDuration` is calculating the difference for each unit somehow 🤷 

Either way when you format this in human readable form, it does not seem at all intuitive, as you would say in natural language that the August 1st is a month from now, instead of 2 months and 2 hours away.

Am I missing an option somewhere to achieve this result?

**Comment by fturmel** [comment]
Months in JS dates are 0-based indexes, so that's why you're seeing a 2 months duration. The month of August is 7, not 8.

As for the output of `formatDuration`, you can control which datetime components you're interested in by using the `format` option. https://date-fns.org/docs/formatDuration

```js
import { formatDuration, intervalToDuration } from "date-fns";

const duration = intervalToDuration({
  start: new Date(),
  end: new Date(2021, 7, 1, 12)
});

const formatOptions = {
  delimiter: ", ",
  format: ["years", "months", "weeks", "days"]
};

const result = formatDuration(duration, formatOptions);

console.log(result); // 1 month
```

Not sure what you're using this for, but it could be worth considering one of the `formatDistance` variants.
https://date-fns.org/docs/formatDistance
https://date-fns.org/docs/formatDistanceStrict
https://date-fns.org/docs/formatDistanceToNow
https://date-fns.org/docs/formatDistanceToNowStrict

```js
import { formatDistanceToNow, formatDistanceToNowStrict } from "date-fns";

const date = new Date(2021, 7, 1, 12);
console.log(formatDistanceToNow(date)); // about 1 month
console.log(formatDistanceToNowStrict(date)); // 1 month
```

**Comment by mikemajara** [comment]
@fturmel that makes total sense now. Thanks for such a verbose answer, much appreciated. I will take a look at those formatDistance functions, they didn't come up when I was looking how to do this.

**Comment by mihaisavezi** [comment]
There is definetely something wrong, or at least unexpected
```
// received
formatDuration(
  {
    years: 2,
    months: 9,
    weeks: 1,
    days: 7,
    hours: 5,
    minutes: 9,
    seconds: 30
  },
  { format: ['months', 'weeks'] }
) === '9 months 1 week'
// expected --> 24 + 9 months = 31 months 1 week
```

what I'm asking the function is to get my duration input and format it in months and weeks, not completely exclude part of the input, correct ?

**Comment by mikemajara** [comment]
That is not the behavior [the tests describe](https://github.com/date-fns/date-fns/blob/4544f0cb1d724bfa7ee846b762bacb8fc303a961/src/formatDuration/test.js#L26). I would assume that `formatDuration` only formats the duration, and it's not concerned about transforming the duration. 🤷

**Comment by mihaisavezi** [comment]
I agree, I noticed the tests. But I'm sure you see where I'm coming from, from a user perspective.
